src/libs/history.c

```diff
@@ -1103,6 +1109,53 @@ static gboolean _changes_tooltip_callback(GtkWidget *widget,
   return show_tooltip;
 }
 
+static gchar *_lib_history_button_label(const dt_dev_history_item_t *item)
+{
+  gchar *label = NULL;
+  if(!item)
+    label = g_strdup("");
+  else if(!item->multi_name[0] || strcmp(item->multi_name, "0") == 0)
+    label = g_strdup(item->module->name());
+  else
+    label = g_strdup_printf("%s • %s", item->module->name(), item->multi_name);
+
+  return label;
+}
+
+static void _lib_history_label_change_callback(gpointer instance, gpointer user_data)
+{
+  dt_lib_module_t *self = (dt_lib_module_t *)user_data;
+  dt_lib_history_t *d = (dt_lib_history_t *)self->data;
+
+  // get corresponding item in history (last item)
+  GList *history = darktable.develop->history;
+  if(g_list_length(history) == 0) return;
+
+  // all vertical boxes
+  GtkContainer *hbox = GTK_CONTAINER(d->history_box);
+  GList *hchildren = gtk_container_get_children(hbox);
+
+  // the box, with num / button&label / on-off
+  GtkContainer *box = GTK_CONTAINER(g_list_nth_data(hchildren, 0));
+  GList *children = gtk_container_get_children(box);
+
+  // the button
+  GtkWidget *widget = (GtkWidget*)g_list_nth_data(children, 1);
+
+  const dt_dev_history_item_t *item = (dt_dev_history_item_t *)(g_list_last(history)->data);
+
+  gchar *label = _lib_history_button_label(item);
+
+  gtk_button_set_label(GTK_BUTTON(widget), label);
```

Comment:
This destroys and recreates the label widget. If instead you use `gtk_label_set_text(GTK_LABEL(lab), label);` on the existing label below, you don't have to set h/xalign and ellipsize again

Reply:
Indeed, fixed.